generator/src/main/java/com/scottlogic/deg/generator/walker/ReductiveDecisionTreeWalker.java

```diff
@@ -45,18 +45,7 @@ public ReductiveDecisionTreeWalker(
 
         visualise(rootNode, initialState);
 
-        //calculate a field to fix and start processing
-        ReductiveConstraintNode reduced = treeReducer.reduce(rootNode, initialState);
-
-        if (reduced == null){
```

Comment:
This check was introduced to prevent generation when the first reduction (now called prune) invalidated the state, i.e. the first fixed field invalidated the tree.

Don't we need to keep this logic?